### PR TITLE
Upgrade virtualan-plugin to 2.5.1

### PR DESCRIPTION
**Pull request by cachescrubber**
# Upgrade virtualan-plugin to 2.5.1

This PR is an attempt to consolidate the virtualan-plugin configuration in the spring generator.

Status: ongoing, WIP. 

Current Status

- virtualan-plugin pulls in springdoc by itself.
- our own integration with swagger-ui and springdoc might conflict with virtualan.
- sample should be upgraded to springdoc.

### PR checklist
 
- [x ] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x ] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [ ] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [ ] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by cachescrubber** [comment]
After upgrading the sample to springdoc, the generated spring boot app won't start:

```
Parameter 0 of method openAPIBuilder in org.springdoc.core.SpringDocConfiguration required a single bean, but 2 were found:
	- apiInfo: defined by method 'apiInfo' in class path resource [org/openapitools/configuration/SpringDocConfiguration.class]
	- customOpenAPI: defined by method 'customOpenAPI' in class path resource [io/virtualan/core/util/ApiDocumentGeneration.class]
```

**Comment by wing328** [comment]
@cachescrubber thanks for the PR.

cc @elans3

**Comment by cachescrubber** [comment]
@wing328 `samples/server/petstore/springboot-virtualan`ist not part of a CI run, it seems.

@elans3 It'll probably mean changes on both side to integrate two Apis with springdoc. IIRC we would need to implement `GroupedOpenApi`.

https://springdoc.org/#how-can-i-define-multiple-openapi-definitions-in-one-spring-boot-project 

Any help too find the right solution is much appreciated.

**Comment by cachescrubber** [comment]
Added `samples/server/petstore/springboot-virtualan` to CI.

**Comment by elans3** [comment]
@cachescrubber.. let me know do I need to fix anything for this #PR

File: modules/openapi-generator/src/main/resources/JavaSpring/libraries/spring-boot/pom.mustache [hunk 1]
```diff
@@ -208,13 +211,12 @@
...
             <groupId>io.virtualan</groupId>
             <artifactId>virtualan-plugin</artifactId>
-            <version>2.2.1</version>
+            <version>${virtualan.version}</version>
         </dependency>
         <!-- HyperSQL DB -->
         <dependency>
             <groupId>org.hsqldb</groupId>
             <artifactId>hsqldb</artifactId>
-            <version>2.5.0</version>
```
[the remark this answers is not in the document]
Comment (elans3) [reply modules/openapi-generator/src/main/resources/JavaSpring/libraries/spring-boot/pom.mustache:217]: perfect!

**Comment by wing328** [comment]
@elans3 some jobs failed. Can you please take a look? 

https://github.com/OpenAPITools/openapi-generator/runs/6570573836?check_suite_focus=true

**Comment by cachescrubber** [comment]
CI Failure is caused by https://github.com/OpenAPITools/openapi-generator/pull/12429#issuecomment-1133925097

**Comment by elans3** [comment]
@cachescrubber 
virtualan-plugin version 2.5.2 support 6.0.0 generator version. Please update accordingly

**Pull request merged**